smoke-tests-otel-starter/spring-boot-4/build.gradle.kts

```diff
@@ -26,6 +26,7 @@ dependencies {
 
   testImplementation("org.springframework:spring-test:7.0.2")
   testImplementation("org.springframework.boot:spring-boot-resttestclient")
+  testImplementation("org.springframework.boot:spring-boot-restclient")
```

Comment:
this is unrelated to the other changes, they [released 4.0.1 ](https://github.com/spring-projects/spring-boot/releases/tag/v4.0.1)yesterday which broke stuff.  See https://github.com/spring-projects/spring-boot/issues/48253